### PR TITLE
MAGE-456: Improve Transaction Status Handling

### PR DESCRIPTION
**Pull request by p1fb**

### CODE REVIEW

File: app/code/community/Payone/Core/Model/Service/TransactionStatus/Execute.php [hunk 1]
```diff
@@ -198,4 +262,21 @@ public function getMaxExecutionTime()
...
-}
\ No newline at end of file
+
+    /**
+     * Handles all transaction status that were failed this cron job run
+     * and sends a reporting email to the configured recipient.
+     */
+    private function handleFailed()
+    {
+        $this->getFactory()->helperEmail()->send(
```
Comment (sprankhub) [inline app/code/community/Payone/Core/Model/Service/TransactionStatus/Execute.php]: Should we first check if there are any failed IDs?

**Comment by sprankhub** [comment]
This looks quite promising - thanks!

File: app/code/community/Payone/Core/Model/Service/TransactionStatus/Execute.php (hunk 1 above)
Comment (p1fb) [reply app/code/community/Payone/Core/Model/Service/TransactionStatus/Execute.php]: Thank you for the hint! Ofc we should do this :)

**Comment by BoulangerV** [comment]
Hello,
A comment regarding "This letter cannot be sent" error during sending.

To me, after investigation, it seems that the new email template is not registered into the config.
So some preparation steps are skipped, then the subject is not processed, and finally the validation fails before sending.

**FIX**
1- Before this line : 
	https://github.com/PAYONE-GmbH/magento-1/blob/cfbfcaae10affbd01bbf63da023aa6399ea06c88/app/code/community/Payone/Core/etc/config.xml#L358

we have to add :
`
<payone_transaction_status_error_report translate="label" module="payone_core">
    <label>Error during transactionstatus handling</label>
    <file>payone/core/transaction_status_error_report.phtml</file>
    <type>html</type>
</payone_transaction_status_error_report>`

2- And on line :
	https://github.com/PAYONE-GmbH/magento-1/blob/cfbfcaae10affbd01bbf63da023aa6399ea06c88/app/code/community/Payone/Core/Model/Service/TransactionStatus/Execute.php#L305

We should replace 
	"transaction_status_error_report" 
with 
	"payone_transaction_status_error_report"

Then it should be ok to try the email sending. 
No clue if the sending will go good afterwards, has to be tested :)

Cheers,
Vincent

**Comment by sprankhub** [comment]
What is the current status here? I would love to see this being merged ASAP :)

**Comment by hreinberger** [comment]
This PR is currently queued for testing by our QA. I've pushed it a bit higher since it's already quite old.

**Pull request merged**